data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -125,9 +129,15 @@ public boolean isStoreAvailable() {
     if (currentSlot.isGreaterThan(slot)) {
       throw new IllegalArgumentException(CANNOT_PRODUCE_HISTORIC_BLOCK);
     }
+    return validatorApiChannel.createUnsignedBlock(slot, randao, graffiti, false);
+  }
 
-    return validatorApiChannel
-        .createUnsignedBlock(
+  public SafeFuture<Optional<BeaconBlock>> getUnsignedBeaconBlockAtSlot(
+      UInt64 slot, BLSSignature randao, Optional<Bytes32> graffiti) {
+    if (randao == null) {
+      throw new IllegalArgumentException(NO_RANDAO_PROVIDED);
```

Comment:
can we remove this check? it is done in the called method in any case, right?

Reply:
i started without it but then the conversion causes issues, so putting this back was the easy option...

Reply:
yeah, true.